dissect/target/tools/query.py

```diff
@@ -45,6 +60,45 @@ def record_output(strings=False, json=False):
     return RecordStreamWriter(fp)
 
 
+def wildcard(funclist: List[str], internal_funcs: List[str]) -> (List[str], List[str]):
+    pluginlist = []
+
+    # List of functions (includes implied)
+    allfunc = []
+    # Subset of functions implied by wildcard (these are not allowed to error)
+    implied = []
+
+    # Collect all available plugins, even special ones
+    for available in plugins():
+        pluginlist.append(available)
+
+    for available in _special_plugins("_os"):
+        pluginlist.append(available)
+
+    for funcname in funclist:
+        # Either wildcard or dot notation (enables full path notation for plugin)
+        if funcname.count(".") or funcname[-1:] == "*":
```

Comment:
I will look into this, thanks for the suggestion!

Reply:
Thank you for your feedback. Such a serious consideration deserves serious thought and a serious reply. So I have pondered about your reasoning and took some time to formulate my thoughts about it.

To me this approach feels rather complex (but I am heavily biased towards minimalism and simplicity and I actually tend to err on the simplicity side of things, sometimes I oversimplify which creates new problems - so this might be happening right now). I think the current solution is currently better in terms of simplicity and robustness:

- It does not affect any other parts of the system/internal functions (as far as I can see, but maybe I am wrong), it just normalizes output on-the-fly upon CLI invocation, so I do not see the point of criticism that the currently proposed solution would change the inner return values, as far as I can see this system only resides on the command line interface level
- It does not have any impact on the overall architecture and does not require any knowledge by internal components of the call side (unless I am mistaken, which of course is very possible, feel free to point it out) - which is one of the guiding principles of the project as I understand, i.e. keep the Python API simple, straight-forward (@Schamper). Having to make plugins aware of call-side output preferences would increase complexity of the API significantly, especially if you are developing a new plugin.
- The record type issue does not seems to make sense to me at the moment. Since the types are specified, any consuming system would be able to separate the records into batches, silos or tables. I expect users that rely on bulk data to feed the result into a consuming system instead of manually inspecting it (but even then, the types are clearly visible per record). Only if you would naively gobble up all data in a CSV this would become an issue, but that seems a rather strange thing to do.
- There is a little overlap in terms of functionality with API-related calls, but this is only on a theoretical level, there is not really any duplicate code, I think it would be more complex to DRY this code further, but a certain degree of integration might be possible. It's also worth taking into account why we pursue DRY in the first place, to mitigate complexity and reduce duplication bugs, in this case I think they do not pose a problem.

But of course this is my, heavily biased view, feel free to point out any errors in my thought process.